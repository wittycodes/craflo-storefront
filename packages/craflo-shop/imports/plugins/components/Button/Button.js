"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@material-ui/core");

var useStyles = (0, _core.makeStyles)(function (theme) {
  return {
    buttonProgress: {
      marginLeft: theme.spacing()
    },
    containedPrimary: {
      "color": theme.palette.primary.contrastText,
      "backgroundColor": theme.palette.colors.red,
      "&:hover": {
        "backgroundColor": theme.palette.colors.redHover,
        // Reset on touch devices, it doesn't add specificity
        "@media (hover: none)": {
          backgroundColor: theme.palette.colors.redHover
        }
      }
    },
    outlinedPrimary: {
      "color": theme.palette.colors.red,
      "border": "1px solid ".concat(theme.palette.colors.red),
      "&:hover": {
        "border": "1px solid ".concat(theme.palette.colors.redBorder),
        "backgroundColor": theme.palette.colors.redBackground,
        // Reset on touch devices, it doesn't add specificity
        "@media (hover: none)": {
          backgroundColor: "transparent"
        }
      }
    }
  };
});
/**
 * @name Button
 * @param {Object} props Component props
 * @returns {React.Component} A React component
 */

var Button = _react.default.forwardRef(function Button(props, ref) {
  var children = props.children,
      color = props.color,
      disabled = props.disabled,
      isWaiting = props.isWaiting,
      otherProps = (0, _objectWithoutProperties2.default)(props, ["children", "color", "disabled", "isWaiting"]);
  var classes = useStyles();

  if (color === "error") {
    return /*#__PURE__*/_react.default.createElement(_core.Button, (0, _extends2.default)({
      classes: {
        containedPrimary: classes.containedPrimary,
        outlinedPrimary: classes.outlinedPrimary
      },
      color: "primary",
      disabled: disabled || isWaiting,
      ref: ref
    }, otherProps), children, isWaiting && /*#__PURE__*/_react.default.createElement(_core.CircularProgress, {
      size: 16,
      className: classes.buttonProgress
    }));
  }

  return /*#__PURE__*/_react.default.createElement(_core.Button, (0, _extends2.default)({
    color: color,
    disabled: disabled || isWaiting,
    ref: ref
  }, otherProps), children, isWaiting && /*#__PURE__*/_react.default.createElement(_core.CircularProgress, {
    size: 16,
    className: classes.buttonProgress
  }));
});

Button.propTypes = {
  /**
   * The content of the Button
   */
  children: _propTypes.default.node,

  /**
   * Override or extend the styles applied to the component.
   */
  classes: _propTypes.default.object,

  /**
   * Options: `default` | `inherit` | `primary` | `secondary` | `error`
   */
  color: _propTypes.default.string,

  /**
   * If `true`, the button will be disabled.
   */
  disabled: _propTypes.default.bool,
  // eslint-disable-line

  /**
   * If `true`, the CircularProgress will be displayed and the button will be disabled.
   */
  isWaiting: _propTypes.default.bool,

  /**
   * onClick callback
   */
  onClick: _propTypes.default.func
};
var _default = Button;
exports.default = _default;