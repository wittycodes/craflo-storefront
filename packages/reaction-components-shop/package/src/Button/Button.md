### Overview

Buttons allow users to take an action such as add, save, cancel or delete. Buttons should clearly and simply communicate the action that will happen when they are pressed.

```jsx noeditor
<div style={{ marginBottom: "2rem", border: "1px solid #e6e6e6", padding: "1rem"}}>
  <div style={{ display: "flex", flexWrap: "wrap", justifyContent: "space-evently", marginBottom: "1rem" }}>
    <div style={{marginRight: "1rem"}}>
      <Button variant="contained" color="primary">Solid</Button>
    </div>
    <div style={{marginRight: "1rem"}}>
      <Button variant="outlined" color="primary">Outlined</Button>
    </div>
    <div style={{marginRight: "1rem"}}>
      <Button>Text-only</Button>
    </div>
    <div style={{marginRight: "1rem"}}>
      <Button variant="contained" color="error">Error</Button>
    </div>
    <div style={{ marginRight: "1rem" }}>
      <Button variant="outlined" color="error">Error</Button>
    </div>
  </div>
  <div style={{ display: "flex", flexWrap: "wrap", justifyContent: "space-evently"}}>
    <div style={{marginRight: "1rem"}}>
      <Button size="small" variant="contained" color="primary">Solid</Button>
    </div>
    <div style={{marginRight: "1rem"}}>
      <Button size="small" variant="outlined" color="primary">Outlined</Button>
    </div>
    <div style={{marginRight: "1rem"}}>
      <Button size="small">Text-only</Button>
    </div>
    <div style={{marginRight: "1rem"}}>
      <Button size="small" variant="contained" color="error">Error</Button>
    </div>
    <div style={{ marginRight: "1rem" }}>
      <Button size="small" variant="outlined" color="error">Error</Button>
    </div>
  </div>
</div>
```

### Usage

#### Types

##### Contained buttons

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button variant="contained" color="primary">Contained</Button>
  </div>
</div>
```

##### Outlined buttons

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button variant="outlined" color="primary">Outlined</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button variant="outlined" color="error">Outlined</Button>
  </div>
</div>
```

##### Error buttons

The error button is used for a destructive action that is difficult to recover from such as deleting information. The error button should be used at the point that the destructive action actually takes place.

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button variant="contained" color="error">Error</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button variant="outlined" color="error">Error</Button>
  </div>
</div>
```

##### Text buttons

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button variant="text" color="primary">Text</Button>
  </div>
</div>
```

##### Small buttons

```jsx
<div style={{ display: "flex" }}>
  <div style={{ marginRight: "1rem" }}>
    <Button size="small" variant="contained" color="primary">Button</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button size="small" variant="outlined" color="primary">Button</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button size="small" variant="outlined" color="error" size="small">Error - Small</Button>
  </div>
  <div style={{ marginRight: "1rem" }}>
    <Button variant="text" size="small">Text - Small</Button>
  </div>
</div>
```

#### Example

##### Button groups

- Follow these patterns for groups of buttons in Cards and the Primary AppBar:

```jsx
<div>
  <Button style={{marginRight: "0.5rem"}} variant="text" color="primary">Discard</Button>
  <Button style={{marginRight: "0.5rem"}} variant="outlined" color="primary">Save changes</Button>
  <Button style={{marginRight: "0.5rem"}} variant="contained" color="primary">Publish</Button>
</div>
```

```jsx
<div>
  <Button style={{marginRight: "0.5rem"}} variant="outlined" color="primary">Cancel</Button>
  <Button style={{marginRight: "0.5rem"}} variant="contained" color="primary">OK</Button>
</div>
```


```jsx
<div>
  <Button style={{marginRight: "0.5rem"}} variant="outlined" color="primary">Cancel</Button>
  <Button style={{marginRight: "0.5rem"}} variant="contained" color="primary">Save</Button>
</div>
```

```jsx
<div>
  <Button style={{marginRight: "0.5rem"}} variant="outlined" color="primary">Cancel</Button>
  <Button style={{marginRight: "0.5rem"}} variant="contained" color="error">Delete</Button>
</div>
```

### API

The Catalyst Button inherits from the Material-UI [Button component](https://material-ui.com/components/buttons/). Refer to the Material-UI [Button API docs](https://material-ui.com/api/button/) for more information.