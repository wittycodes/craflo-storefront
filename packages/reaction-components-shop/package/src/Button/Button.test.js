import React from "react";
import { render } from "../tests/index.js";
import Button from "./Button";

test("basic snapshot - only default props", () => {
  const { asFragment } = render(<Button className="myBtn">Submit</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test("error button snapshot", () => {
  const { asFragment } = render(<Button className="myBtn" color="error" variant="contained">Delete</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test("error button snapshot", () => {
  const { asFragment } = render(<Button className="myBtn" color="error" variant="outlined">Delete</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test("isWaiting button snapshot", () => {
  const { asFragment, getByText } = render(<Button className="myBtn" isWaiting>Upload</Button>);
  expect(getByText("Upload")).toBeDisabled();
  expect(asFragment()).toMatchSnapshot();
});
