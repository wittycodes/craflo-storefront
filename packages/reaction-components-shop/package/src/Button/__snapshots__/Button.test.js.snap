// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot - only default props 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text myBtn"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Submit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`error button snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained myBtn MuiButton-containedPrimary makeStyles-containedPrimary-47"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Delete
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`error button snapshot 2`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined myBtn MuiButton-outlinedPrimary makeStyles-outlinedPrimary-93"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Delete
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`isWaiting button snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text myBtn Mui-disabled Mui-disabled"
    disabled=""
    tabindex="-1"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Upload
      <div
        class="MuiCircularProgress-root makeStyles-buttonProgress-136 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 16px; height: 16px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </span>
  </button>
</DocumentFragment>
`;
